ext/filter/filter.c

```diff
@@ -601,21 +603,22 @@ static void php_filter_call(zval *filtered, zend_long filter, zval *filter_args,
 }
 /* }}} */
 
-static void php_filter_array_handler(zval *input, zval *op, zval *return_value, zend_bool add_empty) /* {{{ */
-{
+static void php_filter_array_handler(zval *input, HashTable *op_ht, zend_long op_long,
+	zval *return_value, zend_bool add_empty
+) /* {{{ */ {
 	zend_string *arg_key;
 	zval *tmp, *arg_elm;
 
-	if (!op) {
+	if (!op_long && !op_ht) {
 		ZVAL_DUP(return_value, input);
-		php_filter_call(return_value, FILTER_DEFAULT, NULL, 0, FILTER_REQUIRE_ARRAY);
-	} else if (Z_TYPE_P(op) == IS_LONG) {
+		php_filter_call(return_value, FILTER_DEFAULT, NULL, 0, 0, FILTER_REQUIRE_ARRAY);
```

Comment:
Rather than have this case, shouldn't the default the argument to `FILTER_DEFAULT` (as it was before, actually)?

Reply:
I'm really sorry, but I don't exactly get you here :( Can you please clarify?